Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Config/MRTKOculusConfig.cs

```diff
@@ -0,0 +1,297 @@
+//copies of the Software, and to permit persons to whom the Software is
+//furnished to do so, subject to the following conditions :
+//
+//The above copyright notice and this permission notice shall be included in all
+//copies or substantial portions of the Software.
+//
+//THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+//AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+//------------------------------------------------------------------------------ -
+
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace prvncher.MixedReality.Toolkit.Config
+{
+#if UNITY_EDITOR
+    [UnityEditor.InitializeOnLoad]
+#endif
+    [CreateAssetMenu(menuName = "MRTK-Quest/MRTK-OculusConfig")]
+    public class MRTKOculusConfig : ScriptableObject
+    {
+        /// <summary>
+        /// Enum used for controlling the teleport pointer activated by MRTK Quest controllers.
+        /// </summary>
+        public enum TeleportPointerMode
+        {
+            Custom,
+            Official,
+            None
+        }
+
+        private static MRTKOculusConfig instance;
+        public static MRTKOculusConfig Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    // Creating a configurable instance so that making any runtime changes won't affect the original. 
+                    MRTKOculusConfig configFile = Resources.Load<MRTKOculusConfig>("MRTK-OculusConfig");
+                    instance = Instantiate(configFile);
+
+                    if (instance == null)
+                    {
+                        UnityEngine.Debug.LogError("Failure to detect MRTK-OculusConfig. Please create an instance using the asset context menu, and place it in any Resources folder.");
+                    }
+                }
+                return instance;
+            }
+        }
+
+        [Header("Config")]
+        [SerializeField]
+        [Tooltip("Using avatar hands requires a local avatar prefab. Failure to provide one will result in nothing being displayed. \n\n" +
+                 "Note: In order to render avatar hands, you will need to set an app id in OvrAvatarSettings. Any number will do, but it needs to be set.")]
+        private bool renderAvatarHandsInsteadOfControllers = true;
+
+        /// <summary>
+        /// Using avatar hands requires a local avatar prefab. Failure to provide one will result in nothing being displayed.
+        /// </summary>
+        public bool RenderAvatarHandsInsteadOfController => renderAvatarHandsInsteadOfControllers;
+
+
+#if OCULUSINTEGRATION_PRESENT
```

Comment:
The platform-conditional serialization thing has historically caused issues:

https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8072

Fixing it is a bit involved, and FWIW I guess I wouldn't want to block on that. However this does add some technical debt. I would just ask that you read that PR and understand the issue at hand here, even though I think fixing it in this PR would isn't necessary.

Reply:
Do we need this file if we're using our own device manager?